File: starter_kits/Scala/hlt/Player.scala
package hlt

import scala.collection.mutable

object Player {
  def initialize: Player = {
    val input = Input.readInput
    val playerId = new PlayerId(input.getInt)
    val shipyard_x = input.getInt
    val shipyard_y = input.getInt
    new Player(playerId, new Shipyard(playerId, new Position(shipyard_x, shipyard_y)))
  }
}

class Player private(val id: PlayerId, val shipyard: Shipyard) {
  final val ships = new mutable.LinkedHashMap[EntityId, Ship]
  final val dropoffs = new mutable.LinkedHashMap[EntityId, Dropoff]
  var halite = 0

  def _update(numShips: Int, numDropoffs: Int, halite: Int) = {
    this.halite = halite
    ships.clear()
    (0 until numShips).foreach({ i =>
      val ship = Ship._generate(id)
      ships.put(ship.id, ship)
    })
    dropoffs.clear()
    (0 until numDropoffs).foreach({ i =>
      val dropoff = Dropoff._generate(id)
      dropoffs.put(dropoff.id, dropoff)
    })
  }

  override def toString() = s"Player{id=${id.id},shipyard=${shipyard},halite=$halite,#ships=${ships.size},#dropoffs=${dropoffs.size}}"
}

File: starter_kits/PHP/hlt/Player.php
<?php

namespace hlt;

class Player
{
    /**
     * @var int
     */
    public $id;

    /**
     * @var Shipyard
     */
    public $shipyard;

    /**
     * @var int
     */
    public $halite;
    /**
     * @var Ship[]
     */
    public $ships = [];
    /**
     * @var Dropoff[]
     */
    public $dropoffs = [];

    private function __construct(PlayerId $id, Shipyard $shipyard)
    {
        $this->id = $id;
        $this->shipyard = $shipyard;
    }

    function _update(int $numShips, int $numDropoffs, int $halite)
    {
        $this->halite = $halite;

        $this->ships = [];
        for ($i = 0; $i < $numShips; ++$i) {
            $ship = Ship::_generate($this->id);

            $this->ships[$ship->id->id] = $ship;
        }

        $this->dropoffs = [];
        for ($i = 0; $i < $numDropoffs; ++$i) {
            $dropoff = Dropoff::_generate($this->id);

            $this->dropoffs[$dropoff->id->id] = $dropoff;
        }
    }

    static function _generate(): Player
    {
        $input = Input::readInput();

        $playerId = new PlayerId($input->getInt());
        $shipyard_x = $input->getInt();
        $shipyard_y = $input->getInt();

        return new Player($playerId, new Shipyard($playerId, new Position($shipyard_x, $shipyard_y)));
    }
}
